frontend/src/pages/CourseSelector/CourseDescription/PrerequisiteTree/PrerequisiteTree.jsx

```diff
@@ -0,0 +1,266 @@
+        },
+        labelCfg: {
+          style: {
+            fill: "#fff",
+            fontFamily: "Arial",
+            fontSize: 14,
+            fontWeight: "bold",
+            cursor: "pointer",
+          },
+        },
+      },
+      defaultEdge: {
+        type: "cubic-horizontal", // polyline could be used but pivots are unreliable with lots of courses
+        color: "#A6A6A6",
+        size: 1,
+        // style: {
+        //   offset: 100,
+        // },
+        labelCfg: {
+          refX: 25,
+          position: 'start',
+          autoRotate: true,
+          style: {
+            fill: "#595959",
+            fontFamily: "Arial",
+            fontWeight: "bold",
+            fontSize: 14,
+            background: {
+              fill: '#ffffff',
+              padding: [2, 2, 2, 2],
+              radius: 5,
+            },
+          },
+        },
+      },
+    });
+
+    setGraph(treeGraphInstance);
+
+    treeGraphInstance.data(graphData);
+
+    updateEdges(treeGraphInstance, graphData);
+    
+    treeGraphInstance.render(graphData);
+    
+    bringLabelsToFront(treeGraphInstance);
+    
+    treeGraphInstance.on("node:click", (event) => {
+      // open new course tab
+      const node = event.item;
+      const { _cfg: { model: { label } } } = node;
+      dispatch(addTab(label));
+    });
+  };
+
+  const updateTreeGraph = (graphData) => {
+    graph.changeData(graphData);
+    bringLabelsToFront(graph);
+  };
+
+  /* MAIN */
+  const setupGraph = async (c) => {
+    setLoading(true);
+    let unlocks = [];
+    let prereqs = [];
+
+    // get all courses a course c unlocks
+    // must pass in no courses 
+    const { startYear } = planner;
+    const { programCode, specs } = degree;
+    const specialisations = {};
+    specs.forEach((spec) => { specialisations[spec] = 1; });
+    
+    const [unlockData, unlockErr] = await axiosRequest(
+      "post",
+      `/courses/coursesUnlockedWhenTaken/${c}`,
+      {
+        program: programCode,
+        specialisations,
+        courses: {},
+        year: new Date().getFullYear() - startYear,
+      },
+    );
+    if (!unlockErr) {
+      // only use direct unlocks
+      unlocks = unlockData.direct_unlock;
+      setCoursesPathTo(unlocks);
+    }
+
+    // get prereqs for course c
+    const [prereqData, prereqErr] = await axiosRequest(
+      "get",
+      `/courses/getPathFrom/${c}`,
+    );
+    if (!prereqErr) {
+      prereqs = prereqData.courses;
+      setCoursesPathFrom(prereqs);
+    }
+
+    // create graph data
+    const graphData = {
+      id: 'root',
+      label: currCourse,
+      children: prereqs?.map((child) => (handleNodeData(child, relationship.PREREQ)))
+        .concat(unlocks?.map((child) => (handleNodeData(child, relationship.UNLOCKS))))
+        .filter(child => child),
+    };
+
+    // render graph
+    if (!graph) {
+      generateTreeGraph(graphData);
+    } else {
+      updateTreeGraph(graphData);
```

Comment:
In what cases will the tree graph update to call this function?

Reply:
oo i just put this here for dev because hot reload kept creating a new graph canvas with every edit HAHAHAHA